src/test/java/io/supertokens/test/userRoles/UserRolesStorageTest.java

```diff
@@ -141,6 +141,13 @@ public void testCreatingAndAddingAPermissionToARoleInTransactionAndDeletingRole(
         thread1.join();
         thread2.join();
 
+        // check that the addPermissionToRole_transaction and deleteRole correctly ran
+        assertTrue(r1_success.get());
+        assertTrue(r2_success.get());
+
+        // check that the transaction in r1 runs once
+        assertEquals(1, numberOfIterations.get());
```

Comment:
```suggestion
        // check that the transaction in r1 runs once. This happens cause it's =
        // running in serialisable transaction isolation level. If we run it in 
        // repeatable read, it will happen twice.
        assertEquals(1, numberOfIterations.get());
```